crates/bin/pcli/src/command/view/noble_address.rs

```diff
@@ -22,31 +36,243 @@ impl NobleAddressCmd {
+        let noble_address = address.noble_forwarding_address(&channel);
 
-        let noble_address = address.noble_forwarding_address(&self.channel);
+        println!("next one-time use Noble forwarding address for account {} is: {}\n\nplease deposit funds to this address...\n\nawaiting deposit...\n\n", account, noble_address);
 
-        println!("{}", noble_address);
+        wait_for_noble_deposit(&noble_node, &noble_address, &address, &channel).await?;
+        println!(
+                    "💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫💫\n\nregistered Noble forwarding account with address {} to forward to Penumbra address {}...\n\nyour funds should show up in your Penumbra account shortly",
+                    noble_address, address
+                );
 
         Ok(())
     }
 }
+
+async fn get_next_noble_sequence(
+    account: Option<u32>,
+    fvk: &FullViewingKey,
+    channel: &str,
+    noble_node: &Url,
+) -> Result<u16> {
+    // perform binary search to find the first unused noble sequence number
+    // search space (sequence number) is 2 bytes wide
+    let left = 0u16;
+    let right = 0xffffu16;
+    let mid = (left + right) / 2u16;
+
+    // attempt to register midpoint
+    _get_next_noble_sequence(left, right, mid, noble_node, channel, fvk, account).await
+}
+
+// Helper function to perform recursive binary search
+fn _get_next_noble_sequence<'a>(
+    left: u16,
+    right: u16,
+    mid: u16,
+    noble_node: &'a Url,
+    channel: &'a str,
+    fvk: &'a FullViewingKey,
+    account: Option<u32>,
+) -> BoxFuture<'a, Result<u16>> {
+    async move {
+        let address = get_forwarding_address_for_sequence(mid, account, fvk);
+        let noble_address = address.noble_forwarding_address(channel);
+        let noble_res =
+            register_noble_forwarding_account(noble_node, &noble_address, &address, channel)
+                .await?;
+        match noble_res {
+            NobleRegistrationResponse::NeedsDeposit => {
+                if left == mid || right == mid {
+                    // We've iterated as far as we can, the next sequence number
+                    // should be the midpoint.
+                    return Ok(mid);
+                }
+
+                // This means the midpoint has not been registered yet. Search the left-hand
+                // side.
+                _get_next_noble_sequence(
+                    left,
+                    mid,
+                    (left + mid) / 2,
+                    noble_node,
+                    channel,
+                    fvk,
+                    account,
+                )
+                .await
+            }
+            NobleRegistrationResponse::Success => {
+                // This means the midpoint had a deposit in it waiting for registration.
+                // This will "flush" this unregistered address, however the user still wants a new one, so return the midpoint + 1.
+                Ok(mid + 1)
```

Comment:
What about these cases:
- Both sequence 23 and sequence 24 has funds in it. Binary search ends up hitting 23 first with a registration call. It succeeds in flushing the funds and then this function returns 24. We display that address to the user, but when we poll, we successfully flush that one too. Then the user attempts to deposit into that address. Will that deposit bounce as it's already registered? 
- What if we flush the max sequence number. Would this return max + 1?

Reply:
1. The program should check the address before it is displayed to ensure it's fresh (could have been used by another instance on another machine)

2. Yes it should wrap around after 2^16 deposits

Reply:
Re point 2:

With a binary search, how there could be a notion of wrapping around? All sequence numbers will have been exhausted. Or are you suggesting once all are registered, we just offer up sequence 0 to re-use?

Reply:
Wrap-around is hard to do without state tracking, so I think using a random sequence number (and potentially notifying the user that it's being re-used) is a good idea